---
title: "SARS-CoV-2 Genomic Sequencing Report"
runtime: shiny
output:
  html_document:
    css: "wslh-theme/wslh-style.css"
    template: "wslh-theme/wslh-theme.html"

---

```{r global, include=FALSE}
library(rjson)
library(tidyr)
library(plotly)
library(viridis)
library(dplyr)
library(usmap)
library(stringr)

# load data in 'global' chunk so it can be shared by all users of the dashboard
# this helps for faster loading!
sc2Data = read.csv(Sys.glob(file.path('./data/gisaid_*tsv')),sep="\t")
dhsdata = read.csv("./data/County_Table_data.csv")
# get geojson data for counties in the US
WICounty_geojson <- fromJSON(file="./data/geojson-counties-fips.json")
herc_geojson <- fromJSON(file="./data/Wisconsin_Healthcare_Emergency_Readiness_Coalition_Regions.json")

source("sc2_variant_freq.R")
source("sequenceMap.R")
source("county_to_herc.R")
```
<div style="margin-bottom:50px">
# SARS-CoV-2 Sequencing Report
</br>
The data available in this report was obtained from the <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> database and the Wisconsin Department of Health Services SARS-CoV-2 dashboard. For additional SARS-CoV-2 information in Wisconsin visit the dashboard here [COVID-19: Wisconsin Summary Data](https://www.dhs.wisconsin.gov/covid-19/data.htm). The data summarized in this report was the result of a combined effort between [WSLH](http://www.slh.wisc.edu/) and its academic, clinical, and public health partners including: [DHS](https://www.dhs.wisconsin.gov/), [UW-Madison AIDS Vaccine Research Laboratory](https://dholk.primate.wisc.edu/wiki/home/page.view?name=home_index), [Gundersen Medical Foundation](https://www.gundersenhealth.org/foundation/), [City of Milwaukee Health Department Laboratory](https://city.milwaukee.gov/Health/Services-and-Programs/healthlab), and the [CDC](https://www.cdc.gov/).
</div>

-----------------------------------------------------------------------
<div style="margin-bottom:50px">
### SARS-CoV-2 Genomes from Wisconsin {.tabset}

#### Cumulative Total

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}

renderPlotly(renderTotal(sc2Data))

```
Cumulative number of sequences originating from Wisconsin patients in <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> by date of sample collection. 


#### Total by Timeframe

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}
selectInput(inputId="totaltimeframe",label="Time Period",choices=c("Weekly","Monthly","Quarterly"))
# load SC2 data
sc2byDate <- data.frame(table(sc2Data$Collection.date))
names(sc2byDate) <- c("date","num")
sc2byDate <- sc2byDate[!(sc2byDate$date=="2020"|sc2byDate$date=="2021"),]
sc2byDate$date <- as.Date(sc2byDate$date, format= "%Y-%m-%d")
sc2byDate <- within(sc2byDate, {
  weeks <- format(date, "%U-%Y")
  weeks <- factor(weeks, levels = unique(weeks))
  
  months <- format(date, "%B-%Y")
  months <- factor(months, levels = unique(months))
  
  quarters <- paste(quarters(date), format(date, "%Y"), sep = "-")
  quarters <- factor(quarters, levels = unique(quarters))
})
sc2byDate <- sc2byDate[!is.na(sc2byDate$date),]
sc2byDate <- droplevels(sc2byDate)

weekly <- sc2byDate %>% group_by(weeks) %>% summarise(num = sum(num))
names(weekly) <- c("date","num")
monthly <- sc2byDate %>% group_by(months) %>% summarise(num = sum(num))
names(monthly) <- c("date","num")
quarterly <- sc2byDate %>% group_by(quarters) %>% summarise(num = sum(num))
names(quarterly) <- c("date","num")


renderPlotly({
  if(input$totaltimeframe == "Monthly"){
    renderTimeTotal(monthly)
  }
  else if(input$totaltimeframe == "Quarterly"){
    renderTimeTotal(quarterly)
  }
  else{
    renderTimeTotal(weekly)
  }
  })
```
Number of sequences originating from Wisconsin patients in <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> by week of sample collection. 
</div>
-----------------------------------------------------------------------

### SARS-CoV-2 Variants {.tabset}

#### Variants of Concern

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}
renderPlotly(renderVOC(sc2Data))
```
**B.1.1.7** - Also known as 20I/501Y.V1 was initially found in December 2020 and first identified in the United Kingdom. This variant carries 17 defining mutations and has spread globally. Early evidence suggests the variant may be associated with increased transmissibility and risk of death. Notable mutations include Spike: N501Y. More info: [B.1.1.7](https://outbreak.info/situation-reports?pango=B.1.1.7)

**B.1.351** - Also known as 20H/501Y.V2 was initially found in December 2020 and first identified in South Africa. This variant carries 9 defining mutations. Some preliminary evidence from non-peer reviewed studies suggest this variant could affect vaccine effectiveness. Notable mutations include Spike: N501Y, Spike: E484K, and Spike: K417T. More info: [B.1.351](https://outbreak.info/situation-reports?pango=B.1.351)

**B.1.429 & B.1.427** - Also known as (CAL.20C) and first identified in Southern California in July 2020. Carries the Spike: L452R mutation which could affect antibody binding [Liu et al. 2020](https://www.biorxiv.org/content/10.1101/2020.11.06.372037v1). More info: [B.1.427](https://outbreak.info/situation-reports?pango=B.1.427) [B.1.429](https://outbreak.info/situation-reports?pango=B.1.429)

**P.1** - Also known as 20J/501Y.V3 was initially found in January 2021 and first identified in Brazil. This variant carries 16 defining mutations. Some preliminary evidence from non-peer reviewed studies suggest this variant could affect vaccine effectiveness. Notable mutations include Spike: N501Y, Spike: E484K, and Spike: K417T. More info: [P.1](https://outbreak.info/situation-reports?pango=P.1)

Read more about these variants at [CDC Variants of Concern](https://www.cdc.gov/coronavirus/2019-ncov/cases-updates/variant-surveillance/variant-info.html#Concern)

#### Variants of Interest

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}
renderPlotly(renderVOI(sc2Data))
```

**P.2** - First identified in Brazil, carries the Spike: E484K mutation but is missing the Spike: N501Y and Spike: K417T. More info: [P.2](https://outbreak.info/situation-reports?pango=P.2)

**B.1.525** - Similar to P.1 and P.2 carries the Spike: E484K mutation. More info: [B.1.525](https://outbreak.info/situation-reports?pango=B.1.525)

**B.1.526** - First identified in New York. More info: [B.1.526](https://outbreak.info/situation-reports?pango=B.1.526)

Read more about these variants at [CDC Variants of Interest](https://www.cdc.gov/coronavirus/2019-ncov/cases-updates/variant-surveillance/variant-info.html#Interest)

#### Proportion of Variants Sequenced
```{r, echo=FALSE,message=FALSE,warning=FALSE,fig.align="center", out.width="80%"}
selectInput(inputId="vartimeframe",label="Time Period",choices=c("Weekly","Monthly","Quarterly"))
sc2bylineage <- data.frame(table(sc2Data$Collection.date,sc2Data$Lineage))
names(sc2bylineage) <- c("date","lineage","num")
sc2bylineage <- sc2bylineage[!(sc2bylineage$date=="2020"|sc2bylineage$date=="2021"),]
sc2bylineage$date <- as.Date(sc2bylineage$date, format= "%Y-%m-%d")
sc2bylineage <- within(sc2bylineage, {
  weeks <- format(date, "%U-%Y")
  weeks <- factor(weeks, levels = unique(weeks))
  
  months <- format(date, "%B-%Y")
  months <- factor(months, levels = unique(months))
  
  quarters <- paste(quarters(date), format(date, "%Y"), sep = "-")
  quarters <- factor(quarters, levels = unique(quarters))
})
sc2bylineage <- sc2bylineage[!is.na(sc2bylineage$date),]
sc2bylineage <- droplevels(sc2bylineage)
sc2bylineage <- sc2bylineage %>% mutate_at(c("date","quarters","months","weeks","lineage"), as.character())

varWeekly <- group_by_at(sc2bylineage,vars(weeks,lineage)) %>% summarise(num = sum(num))
names(varWeekly) <- c("date","lineage","num")
varMonthly <- group_by_at(sc2bylineage,vars(months,lineage)) %>% summarise(num = sum(num))
names(varMonthly) <- c("date","lineage","num")
varQuarterly <- group_by_at(sc2bylineage,vars(quarters,lineage)) %>% summarise(num = sum(num))
names(varQuarterly) <- c("date","lineage","num")

#keep only top 5 (weekly)
df <- data.frame(matrix(ncol = 3, nrow = 0))
x <- c("date","lineage","num")
colnames(df) <- x
for(week in unique(varWeekly$date)){
  dataHolder <- varWeekly[varWeekly$date == week,]
  dataHolder <- dataHolder[order(dataHolder$num,decreasing = TRUE),]
  a <- dataHolder[1:5,]
  a <- a %>% mutate_at(c("date","lineage"), as.character())
  b <- data.frame("date"=week,"lineage"="other","num"=sum(dataHolder[6:nrow(dataHolder),]$num),stringsAsFactors = FALSE)
  a <- rbind(a,b)
  df <- rbind(df,a)
}
varWeekly <- df

#keep only top 5 (monthly)
df <- data.frame(matrix(ncol = 3, nrow = 0))
x <- c("date","lineage","num")
colnames(df) <- x
for(month in unique(varMonthly$date)){
  dataHolder <- varMonthly[varMonthly$date == month,]
  dataHolder <- dataHolder[order(dataHolder$num,decreasing = TRUE),]
  a <- dataHolder[1:5,]
  a <- a %>% mutate_at(c("date","lineage"), as.character())
  b <- data.frame("date"=month,"lineage"="other","num"=sum(dataHolder[6:nrow(dataHolder),]$num),stringsAsFactors = FALSE)
  a <- rbind(a,b)
  df <- rbind(df,a)
}
varMonthly <- df

#keep only top 5 (quarterly)
df <- data.frame(matrix(ncol = 3, nrow = 0))
x <- c("date","lineage","num")
colnames(df) <- x
for(quarter in unique(varQuarterly$date)){
  dataHolder <- varQuarterly[varQuarterly$date == quarter,]
  dataHolder <- dataHolder[order(dataHolder$num,decreasing = TRUE),]
  a <- dataHolder[1:5,]
  a <- a %>% mutate_at(c("date","lineage"), as.character())
  b <- data.frame("date"=quarter,"lineage"="other","num"=sum(dataHolder[6:nrow(dataHolder),]$num),stringsAsFactors = FALSE)
  a <- rbind(a,b)
  df <- rbind(df,a)
}
varQuarterly <- df


renderPlotly({
  if(input$vartimeframe == "Monthly"){
    renderTimeLinage(varMonthly)
  }
  else if(input$vartimeframe == "Quarterly"){
    renderTimeLinage(varQuarterly)
  }
  else{
    renderTimeLinage(varWeekly)
  }
  })
```

Plot of the proportion of sequence lineages in <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> by the selected time period. Hover over the plot see more detailed information.

#### Variants by HERC region

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}
renderPlotly(renderHERCMap(sc2Data,dhsdata,herc_geojson))
```
Variants of concern in <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> by [HERC](https://www.dhs.wisconsin.gov/preparedness/healthcare/index.htm) region, lighter colors represent a larger number of total variants. Hover over a region to see a breakdown of the variants of concern.
  
###

-----------------------------------------------------------------------
### SARS-CoV-2 Genomes by County

```{r, echo=FALSE,message=FALSE,fig.align="center", out.width="80%"}
renderPlotly(renderCountyMap(sc2Data,dhsdata,WICounty_geojson))
```
Total number of sequences in <a href="https://www.gisaid.org/"> <img src="https://www.gisaid.org/fileadmin/gisaid/img/schild.png" alt="GISAID" width="50"/></a> by County, lighter colors represent a larger number of total sequences. Hover over a County to see what percentage of confirmed cases have been sequenced. Confirmed cases by county are available [here](https://www.dhs.wisconsin.gov/covid-19/county.htm).
