---
title: "SARS-CoV-2 Genomic Sequencing Report"
runtime: shiny
output:
  html_document:
    css: "wslh-theme/wslh-style.css"
    template: "wslh-theme/wslh-r-template.html"
---

```{r global, include=FALSE}
# load data in 'global' chunk so it can be shared by all users of the dashboard
# this helps for faster loading!
sc2Data = read.csv(Sys.glob(file.path('./data/gisaid_*tsv')),sep="\t")
dhsdata = read.csv("./data/County_Table_data.csv")
```

# SARS-CoV-2 Sequencing Report

-----------------------------------------------------------------------

### SARS-CoV-2 Genomes from Wisconsin

```{r, echo=FALSE,message=FALSE,fig.align="center", fig.width=9.5}
source("sc2_variant_freq.R", local = knitr::knit_global())
renderTotal(sc2Data)
```

-----------------------------------------------------------------------

### SARS-CoV-2 Variants {.tabset}

#### Variants of Concern

```{r, echo=FALSE,message=FALSE,fig.align="center", fig.width=9.5}
source("sc2_variant_freq.R", local = knitr::knit_global())
renderVOC(sc2Data)
```

#### Variants of Interest

```{r, echo=FALSE,message=FALSE,fig.align="center", fig.width=9.5}
source("sc2_variant_freq.R", local = knitr::knit_global())
renderVOI(sc2Data)
```


-----------------------------------------------------------------------

### SARS-CoV-2 Genomes by County

```{r, echo=FALSE,message=FALSE,fig.align="center", fig.width=9.5}
source("sequenceMap.R", local = knitr::knit_global())
renderMap(sc2Data,dhsdata)
```

